src/app/home/home.page.ts

```diff
@@ -74,102 +76,101 @@ export class HomePage implements OnInit {
       }
     });
   }
-  createLetters() {
+
+  private createLetters() {
     this.letters = Object.keys(this.groupedCountries).sort();
   }
 
- 
+  private updatedLoadingState(item: Country): void {
+    this.isLoaded = true;
+    this.isLoading = false;
+    this.loadedItem = item;
+    this.loadingItem = null;
+  }
+
+  private addToFavorites(item: Country): void {
+    if (!this.favourites.includes(item))
+      this.favourites.push(item);
+  }
+
+  private removeFromFavorites(item: Country): void {
+    this.favourites = this.favourites.filter(c => c !== item);
+  }
+
+  private updateView(): void {
+    if (!this.isHomePage) {
+      this.groupedCountries = {};
+      this.countries = this.favourites;
+      this.groupCountries();
+      this.createLetters();
+    } else {
+      this.groupedCountries = this.baseBackupGroupedCountries;
+    }
+  }
+
+  filterCountries(searchBarValue: string | undefined | null): void {
+    const query = searchBarValue?.toLowerCase().trim();
+    if (query && query !== '') {
+      this.groupedCountries = {};
+      this.countries = this.isHomePage ? this.baseCountries.filter(country => country.name.toLowerCase().indexOf(query) > -1)
+        : this.favourites.filter(country => country.name.toLowerCase().indexOf(query) > -1);
+      this.groupCountries();
+    } else {
+      this.updateView();
+    }
+    this.createLetters();
+  }
+
+  toggleView(): void {
+    this.isHomePage = !this.isHomePage;
+    this.filterCountries(this.searchBar.value);
```

Comment:
use safe check

Reply:
@mnehro 
If you want to ensure that this.isHomePage is false when it is null or undefined, you should use the nullish coalescing operator like this:

`this.isHomePage = this.isHomePage ?? false;
`